elastic/kibana
# [Upgrade Assistant] Remove index_mapper_dynamic setting when reindexing

### Summary
As surfaced by https://github.com/elastic/kibana/issues/116212, when updating from 5.5 to 6.20 to 7.16 and attempting to reindex `.reindexed-v6-kibana-6` users get an error saying `Setting index.mapper.dynamic was removed after version 6.0.0`. This is a manual deprecation that needs to be solved before running the reindex process. Turns out that within the UA reindexing code there's a list of blacklisted settings that get removed before the reindexing runs and this particular setting, that was removed with [elasticsearch#51718](https://github.com/elastic/elasticsearch/pull/51718), is not there.

**How to test**
1. Unzip [5.5-data-snapshot.zip](https://github.com/elastic/kibana/files/7561846/5.5-data-snapshot.zip) and start up ES as follows: `yarn es snapshot -E path.data=/path/to/5.5-data-snapshot`. This contains indices from 5.5 that where migrated to 6.8 and then to 7.16.
4. Navigate to `Stack Management` -> `Upgrade Assistant` -> `ES deprecation issues`
5. Verify that for `.reindexed-v6-kibana-6` there are two critical deprecations: `Index created before 7.0` and `Setting [index.mapper.dynamic] is deprecated`
6. Open the `Index created before 7.0` deprecation and verify that it can be successfully re-indexed
7. Verify after clicking the `Refresh` CTA from the `ES deprecation issues` page, the `Setting [index.mapper.dynamic] is deprecated` deprecation is gone.

If after following the steps you don't see any deprecations for es in upgrade assistant, try unzipping the folder in the `kibana/.es` folder and starting up es with: `yarn es snapshot -E path.data=../5.5-data-snapshot`

**Screenshots**

<details>
<summary>before/after</summary>

// With error
![138763612-18f03d01-5058-44e5-a160-80438561f794](https://user-images.githubusercontent.com/1191206/142394037-39b16d33-944e-4daf-b2c6-074fbc86beb9.png)

// Solved deprecation
![141445459-42814480-3a15-4e98-a3eb-6b2d0f86446c](https://user-images.githubusercontent.com/1191206/142396821-52ed1484-fb01-413a-af2d-7468f3145420.png)

</details>
